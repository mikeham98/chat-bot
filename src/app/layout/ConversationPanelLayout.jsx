import React, {Suspense} from 'react';
import SideBar from "../components/common/SideBar";
import Spinner from "../components/common/Spinner";
import store from "../store";
import conversationsReducer from "../reducers/conversations/index.reducers";

const ConversationsListContainer = React.lazy(() =>
    import(/* webpackChunkName: "CONVERSATIONS_LIST_CONTAINER" */ "../containers/Conversations/ConversationsListContainer")
        .then(() => {
            store.injectReducer('conversations', conversationsReducer);
            return import ("../containers/Conversations/ConversationsListContainer");
        })
);
export default class ConversationPanelLayout extends React.Component {
    render() {
        return (
            <SideBar title='Messages'>
                <Suspense fallback={<Spinner/>}>
                    <ConversationsListContainer/>
                </Suspense>
            </SideBar>
        )
    }
}