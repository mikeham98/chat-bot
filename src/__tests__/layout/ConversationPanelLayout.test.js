import React from 'react';
import {shallow} from 'enzyme';
import ConversationPanelLayout from '../../app/layout/ConversationPanelLayout';

describe('ConversationPanelLayout', () => {

    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<ConversationPanelLayout {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {};
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return a lazy component', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
});