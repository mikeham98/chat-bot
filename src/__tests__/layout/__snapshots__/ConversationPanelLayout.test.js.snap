// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationPanelLayout render should return a lazy component 1`] = `
<SideBar
  title="Messages"
>
  <Suspense
    fallback={<Memo(Spinner) />}
  >
    <lazy />
  </Suspense>
</SideBar>
`;
